gradle/eclipse-project-layout.gradle

```diff
@@ -3,19 +3,21 @@
  * live in the same folders, we need to make some adjustments to Gradle'
  * defaults.
  */
+ 
+def isTestProject = name.endsWith('tests') 
+
 sourceSets {
-	main {
+	isTestProject? test : main  {
 		java.srcDirs = ['src', 'xtend-gen', 'src-gen']
 		resources {
 			srcDirs = ['src', 'src-gen']
-			exclude (
-				'**/*._trace',
-				'**/*.xtendbin',
-				'**/*.xtend',
-				'**/*.mwe2',
-			)
+			exclude '**/*._trace', '**/*.xtendbin', '**/*.xtend', '**/*.mwe2'
```

Comment:
I think it should be symmetric to the compiler settings in eclipse. There we filter '*.launch,*.xtend,*._trace,*.xtendbin' in org.eclipse.xtext, for example. The settings should be made consistent across all the compiler configs.